modules/asm/README.md

```diff
@@ -2,87 +2,37 @@
 
 This module installs [Anthos Service Mesh](https://cloud.google.com/service-mesh/docs) (ASM) in a Kubernetes Engine (GKE) cluster.
 
-Specifically, this module automates installing the ASM Istio Operator on your cluster ([installing ASM](https://cloud.google.com/service-mesh/docs/install)).
-
 ## Usage
 
-There is a [full example](../../examples/simple_zonal_with_asm) provided. Detailed usage example is as follows:
-
-```tf
-module "asm" {
-  source                = "terraform-google-modules/kubernetes-engine/google//modules/asm"
-
-  project_id            = "my-project-id"
-  cluster_name          = "my-cluster-name"
-  location              = module.gke.location
-  cluster_endpoint      = module.gke.endpoint
-  enable_all            = false
-  enable_cluster_roles  = true
-  enable_cluster_labels = false
-  enable_gcp_apis       = false
-  enable_gcp_iam_roles  = true
-  enable_gcp_components = true
-  enable_registration   = false
-  managed_control_plane = false
-  options               = ["envoy-access-log,egressgateways"]
-  custom_overlays       = ["./custom_ingress_gateway.yaml"]
-  skip_validation       = true
-  outdir                = "./${module.gke.name}-outdir-${var.asm_version}"
-}
-```
-
-To deploy this config:
+There are a few prerequisites to using this module that can be done either through Terraform or manually:
 
-1. Run `terraform apply`
+1. Enable the `mesh.cloud.googleapis.com` service
+1. Enable the `servicemesh` feature on the cluster hub
+1. Register target cluster to the servicemesh-enabled hub
 
-## Requirements
+## Migration
```

Comment:
Lets create an upgrade guide for v20 and move this there.
https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/docs

Reply:
Done.